### DISCUSSION TITLE
How to change the target device in the HSS?

### DISCUSSION THREAD

**Discussion by xiyanni** [polarfire-soc/polarfire-soc-documentation#113] [How to] [answered]
From Github, the target device in the proj.config is MPFS250T_ES. The customer is using MPFS025T, could they use the same HSS project?
What's diference when the customer move the HSS from MPFS250T_ES to MPFS025T?

**Comment by hughbreslin** [comment, marked as answer]
Hi @xiyanni 

Where are you seeing a proj.config file? 

The HSS should be ok once it has the correct XML set up for the custom board. The main thing to make sure of is that the correct board is selected for building / programming (if its not an Icicle Kit it should have its own custom board set up). 

If you're building / programming using the command line you'll need to make sure the correct `Package` and `Die` are selected in the [`envm-wrapper/makefile` ](https://github.com/polarfire-soc/hart-software-services/blob/master/envm-wrapper/Makefile#L107) where you can then run:

    $ make
    $ make program

from the command line. 

If you're building / programming via SoftConsole you need to make sure the boot mode programmer external tool is configured correctly with the correct `die` and `package`:

![image](https://user-images.githubusercontent.com/44862881/177114341-4f417723-ff6b-4074-a0f0-33a864894541.png)

Libero should handle this automatically as the `die` and `package` are configured through the project settings

**Comment by xiyanni** [comment]
I found this file at : hart-software-services\Default\bootmode1\fpgenprogProject\proj.config

**Reply by hughbreslin** [reply]
Ahhh ok, that file is generated as part of the boot mode programmer flow so updating the boot mode programmer settings in the screenshot above changes the generated proj.config file :) 

I changed mine to MPFS250T and I can see that reflected in the file:

![image](https://user-images.githubusercontent.com/44862881/177117715-ae69d08e-d45e-4e8f-bbb3-6fb9a05354dc.png)
